lib/core-integration/src/Test/Integration/Scenario/API/Byron/Migrations.hs

```diff
@@ -469,6 +482,50 @@ spec = describe "BYRON_MIGRATIONS" $ do
             . fmap (view (#amount . #getQuantity))
             . view #outputs
 
+    migrateWallet
+        :: Context
+        -> ApiByronWallet
+        -> [(ApiT Address, Proxy n)]
+        -> IO ()
+    migrateWallet ctx src targets = do
+        (st, _) <- request @ApiWalletMigrationInfo ctx endpointInfo Default Empty
+        when (st == HTTP.status200) $ do -- returns '403 Nothing to Migrate' when done
+            -- 1/ Forget all pending transactions to unlock any locked UTxO
+            (_, txs) <- unsafeRequest @[ApiTransaction n] ctx endpointListTxs Empty
+            forM_ txs $ forgetTxIf ((== ApiT Pending) . view #status)
+
+            -- 2/ Attempt to migrate
+            _ <- request @[ApiTransaction n] ctx endpointMigration Default payload
+
+            -- 3/ Wait "long-enough" for transactions to have been inserted.
+            -- What long enough is rather empirical here, but it must satisfies
+            -- one important criteria: it must be long-enough to assume that all
+            -- transactions are no longer in mempools AND, for those who've been
+            -- inserted, they are now deep enough to not be affected by
+            -- rollbacks.
+            threadDelay (5 * oneSecond)
```

Comment:
Maybe that's actually not long enough.

Reply:
It should take 2 seconds for a tx to become immutable once it gets included in a block, so 5 seconds — sounds — good enough at least
```
slotLength: 0.2
maxKESEvolutions: 90
securityParam: 10
```

Reply:
That's my reasoning, 4s is basically `2k`, so 5 seems already conservative. But that's also assuming that the transaction are inserted _right away_. Anyway, I went for 15, just to hammer it down.

Reply:
_Might_ be an idea to
1. Make the reasoning explicit e.g. `threadDelay $ securityParamDelay + 3 * oneSecond`
2. Not go for 15s unless we know we need to.

such that we get a better understanding of exactly what kind of delay is needed, to apply to other tests.